redcap_bridge/server_elab_interface.py

```diff
@@ -0,0 +1,153 @@
+
+    exp = experiment_api.get_experiment(id)
+
+    return exp
+
+
+def create_template_with_metadata(server_config_json, template_file):
+    """
+    Create a template with metadata.
+
+    Parameters
+    ----------
+    server_config_json: str
+        Path to the json file containing the redcap url, api token and required external modules
+    template_file: str
+        Path to the template you want to create
+
+    Returns
+    -------
+        location_response: location of the new template
+
+    """
+
+    api_client = get_elab_config(server_config_json)
+    template_api = elabapi_python.ExperimentsTemplatesApi(api_client)
+
+    if json.load(open(template_file, 'r')):
+        template = json.load(open(template_file, 'r'))
+        response = template_api.post_experiment_template_with_http_info(body={"title": template['title']})
+        location_response = response[2].get('Location')
+
+        item_id = int(location_response.split('/').pop())
+
+        if template['metadata']:
+            template_api.patch_experiment_template(item_id, body={'metadata': template['metadata']})
+        else:
+            raise ValueError(f'No metadata No metadata in the template')
+    else:
+        raise ValueError(f'Unknown format or template. Valid format is "json".')
+
+    return location_response
+
+
+def create_template_without_metadata(server_config_json, template_file):
+    """
+    Create a simple template with a title without metadata.
+
+    Parameters
+    ----------
+    server_config_json: str
+        Path to the json file containing the redcap url, api token and required external modules
+    template_file: str
+        Path to the template you want to create
+
+    Returns
+    -------
+        location_response: location of the new template
+
+    """
+
+    api_client = get_elab_config(server_config_json)
+    template_api = elabapi_python.ExperimentsTemplatesApi(api_client)
+
+    if json.load(open(template_file, 'r')):
+        template = json.load(open(template_file, 'r'))
+        response = template_api.post_experiment_template_with_http_info(body={"title": template['title']})
+        location_response = response[2].get('Location')
+    else:
+        raise ValueError(f'Unknown format or template. Valid format is "json".')
+
+    return location_response
+
+
+def create_template_with_converted_csv(server_config_json, csv_file, title):
+    """
+    Creating a template with an old csv file.
+
+    Parameters
+    ----------
+    server_config_json: str
+        Path to the json file containing the redcap url, api token and required external modules
+    template_file: str
+        Path to the csv file you want to convert into an Elab version
+    title: str
+        Title of the template created
+
+    Returns
+    -------
+        location_response: location of the new template
+
+    """
+
+    json_conversion_file = conversion_csv_to_json(csv_file)
+    json_conversion_file = json.dumps(json_conversion_file)
```

Comment:
To be more clean you should not reuse the variable `json_conversion_file` and put different contents in there. Maybe there's more telling names that distinguish the two contents here?

Reply:
Yes i'll change the name of these two variables